### ISSUE TITLE
Component re-renders list as empty

### ISSUE THREAD

**Issue by henrik** [phoenixframework/phoenix_live_view#960]
### Environment

* Elixir version (elixir -v): Elixir 1.10.3 (compiled with Erlang/OTP 22)
* Phoenix version (mix deps): 1.5.3
* Phoenix LiveView version (mix deps): 0.13.3
* NodeJS version (node -v): 11.0.0
* NPM version (npm -v): 6.4.1
* Operating system: macOS 
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes (actually did try)

### Actual behavior

With the example LiveView below (a LiveView that passes a param value to a component, which conditionally renders a list), you see items listed on the first "Has stuff" click, but if you click "No stuff" and then back to "Has stuff", it doesn't list any items.

![bug](https://user-images.githubusercontent.com/216/83965572-750f4e00-a8ac-11ea-8b8f-52ebd8bab12b.gif)

See below for repro code.

### Expected behavior

To show items on every click of "Has stuff".

### Details

After going back to "Has stuff" and seeing an empty list, the page source includes e.g.

```
<p data-phx-component="0" id="phx-FhY5x6YUGLIb5QlF-1-0" data-phx-skip=""></p>
<p data-phx-component="0" id="phx-FhY5x6YUGLIb5QlF-2-0" data-phx-skip=""></p>
```

where it previously had

```
<p data-phx-component="0" id="phx-FhY5x6YUGLIb5QlF-1-0">My ID: 1</p>
<p data-phx-component="0" id="phx-FhY5x6YUGLIb5QlF-2-0">My ID: 2</p>
```

The logs (logged in `update/2` in the code below) show that assigns has the list:

```
{:update, "stuff",
 [%RemitWeb.DebugLive.Item{id: 1}, %RemitWeb.DebugLive.Item{id: 2}]}
{:update, "empty", []}
{:update, "stuff",
 [%RemitWeb.DebugLive.Item{id: 1}, %RemitWeb.DebugLive.Item{id: 2}]}
```

The WebSocket message looks like this on the first "Has stuff" click:

> ["4","6","lv:phx-FhY6VNhCcVXOqwwB","phx_reply",{"response":{"diff":{"2":{"0":0},"c":{"0":{"0":"stuff","3":{"d":[[1],[2]],"s":["\n    ","\n  "]}},"1":{"0":"1","s":["<p>My ID: ","</p>"]},"2":{"0":"2","s":1}}}},"status":"ok"}]

And like this on the second "Has stuff" click:

> ["4","12","lv:phx-FhY6VNhCcVXOqwwB","phx_reply",{"response":{"diff":{"2":{"0":0},"c":{"0":{"0":"stuff","3":{"d":[[1],[2]]}}}}},"status":"ok"}]

### Repro

Apologies for the long-ish repro but couldn't manage to make it shorter. If I removed MyComponent and rendered the list directly from DebugLive, the issue went away.

```
defmodule RemitWeb.DebugLive do
  use RemitWeb, :live_view
  alias RemitWeb.DebugLive.ItemComponent
  alias __MODULE__

  defmodule Item do
    defstruct [:id]
  end

  defmodule ItemComponent do
    use RemitWeb, :live_component

    @impl true
    def render(assigns), do: ~L"<p>My ID: <%= @id %></p>"
  end

  defmodule MyComponent do
    use RemitWeb, :live_component

    @impl true
    def mount(socket) do
      socket = assign(socket, filter: "empty", items: [])
      {:ok, socket}
    end

    @impl true
    def update(%{filter: nil}, socket) do
      {:ok, socket}
    end
    def update(%{filter: filter}, socket) do
      socket = assign(socket, filter: filter)
      socket =
        if filter == "empty" do
          assign(socket, items: [])
        else
          assign(socket, items: [%Item{id: 1}, %Item{id: 2}])
        end

      IO.inspect {:update, socket.assigns.filter, socket.assigns.items}
      {:ok, socket}
    end

    @impl true
    def render(assigns) do
      ~L"""
        <p>Current filter: <%= @filter %></p>
        <p>
          Filter by:
          <%= live_patch "Has stuff", to: Routes.live_path(@socket, DebugLive, filter: "stuff") %>
          <%= live_patch "No stuff", to: Routes.live_path(@socket, DebugLive, filter: "empty") %>
        </p>

        <%= for item <- @items do %>
          <%= live_component @socket, ItemComponent, id: item.id %>
        <% end %>
      """
    end
  end

  @impl true
  def render(assigns) do
    ~L"""
      <%= live_component @socket, MyComponent, id: :debug, filter: @filter %>
    """
  end

  @impl true
  def mount(_params, _session, socket) do
    {:ok, socket}
  end

  @impl true
  def handle_params(params, _uri, socket) do
    socket = assign(socket, filter: params["filter"])
    {:noreply, socket}
  end
end
```

**Comment by josevalim** [comment]
@henrik is the issue on the diff over the wire or on the client? You can call `liveView.enableDebug()` to see the diffs!

**Comment by henrik** [comment]
@josevalim Over the wire, I believe – I just edited the issue to include the WebSocket messages :)

**Comment by josevalim** [comment]
I have debugged and the diff over the wire seems to be OK! So this looks like a client side issue. :)

**Comment by josevalim** [comment]
Also fixed by #990.

**Comment by josevalim** [comment]
Closing in favor of the PR.
